test/e2e/reconciliation_test.go

```diff
@@ -130,111 +131,114 @@ var _ = Describe("Reconciliation From OLSConfig CR", Ordered, func() {
-		Expect(err).NotTo(HaveOccurred())
+			By("update models in the OLSConfig CR")
+			err = client.Get(cr)
+			Expect(err).NotTo(HaveOccurred())
 
-		By("check the app deployment generation that should be inscreased")
-		err = client.WaitForDeploymentCondition(deployment, func(dep *appsv1.Deployment) (bool, error) {
-			if dep.Generation <= generation {
-				return false, fmt.Errorf("current generation %d is inferior to observed generation %d", dep.Generation, generation)
+			cr.Spec.OLSConfig.DefaultModel = OpenAIAlternativeModel
+			if !slices.Contains(cr.Spec.LLMConfig.Providers[0].Models, olsv1alpha1.ModelSpec{Name: OpenAIAlternativeModel}) {
+				cr.Spec.LLMConfig.Providers[0].Models = append(cr.Spec.LLMConfig.Providers[0].Models, olsv1alpha1.ModelSpec{Name: OpenAIAlternativeModel})
 			}
-			return true, nil
-		})
-		Expect(err).NotTo(HaveOccurred())
-		generation = deployment.Generation
 
-		By("change LLM token secret reference")
-		err = client.Get(cr)
-		Expect(err).NotTo(HaveOccurred())
+			err = client.Update(cr)
+			Expect(err).NotTo(HaveOccurred())
+			By("wait for the app configmap to be updated")
+			configMap = &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      AppServerConfigMapName,
+					Namespace: OLSNameSpace,
+				},
+			}
+			err = client.WaitForConfigMapContainString(configMap, AppServerConfigMapKey, "default_model: "+OpenAIAlternativeModel)
+			Expect(err).NotTo(HaveOccurred())
 
-		cr.Spec.LLMConfig.Providers[0].CredentialsSecretRef.Name = LLMTokenSecondSecretName
-		err = client.Update(cr)
-		Expect(err).NotTo(HaveOccurred())
+			By("check the app deployment generation that should be inscreased")
+			err = client.WaitForDeploymentCondition(deployment, func(dep *appsv1.Deployment) (bool, error) {
+				if dep.Generation <= generation {
+					return false, fmt.Errorf("current generation %d is inferior to observed generation %d", dep.Generation, generation)
+				}
+				return true, nil
+			})
+			Expect(err).NotTo(HaveOccurred())
+			generation = deployment.Generation
 
-		By("check the app deployment generation that should be inscreased")
-		err = client.WaitForDeploymentCondition(deployment, func(dep *appsv1.Deployment) (bool, error) {
-			if dep.Generation <= generation {
-				return false, fmt.Errorf("current generation %d is inferior to observed generation %d", dep.Generation, generation)
-			}
-			return true, nil
-		})
-		Expect(err).NotTo(HaveOccurred())
+			By("change LLM token secret reference")
+			err = client.Get(cr)
+			Expect(err).NotTo(HaveOccurred())
 
-		By("check the app configmap to contain the new secret volume")
-		err = client.WaitForConfigMapContainString(configMap, AppServerConfigMapKey, path.Join("/etc/apikeys", LLMTokenSecondSecretName))
-		Expect(err).NotTo(HaveOccurred())
+			cr.Spec.LLMConfig.Providers[0].CredentialsSecretRef.Name = LLMTokenSecondSecretName
+			err = client.Update(cr)
```

Comment:
this worth a retry on conflict, too

Reply:
isn't it already wrapped in the retry on conflict?

Reply:
should I split it into mutliple retries?

Reply:
`Expect(err).NotTo(HaveOccurred())` that follows this line will interrupt the testing.
we need return the error to let the retryOnConflict attempt again the get and update combo.